### ISSUE TITLE
redesign failover

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2398]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2394
- Created at 2014-07-30 12:22:23 by [pbrezina](https://pagure.io/user/pbrezina)
- Closed at 2020-03-24 14:10:31 as wontfix
- Assigned to nobody
---

Our AD integration has outgrown current failover code. The failover was not designed to support subdomains and we are running in more and more problems which are harder to fix.

The main problem is that the current failover has one context for the whole sssd_be process which makes it impossible to have different settings for root domain and its subdomains. We should redesign the failover in a way that there will be one separate context per domain.

We may also want to implement additional features like caching mechanism (since global catalog servers are the same for all AD forest).

### Comments

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2014-07-30 12:24:21](https://pagure.io/SSSD/sssd/issue/2394#224160)**

Fields changed

blockedby:  => #2393 #1083

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2014-08-06 09:41:15](https://pagure.io/SSSD/sssd/issue/2394#224161)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD 1.13 beta

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2014-08-06 09:43:39](https://pagure.io/SSSD/sssd/issue/2394#224162)**

Fields changed

rhbz:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2014-09-26 18:50:00](https://pagure.io/SSSD/sssd/issue/2394#224163)**

Fields changed

mark:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2015-02-10 14:37:57](https://pagure.io/SSSD/sssd/issue/2394#224164)**

Fields changed

milestone: SSSD 1.13 beta => SSSD 1.13 backlog

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2015-02-12 20:13:39](https://pagure.io/SSSD/sssd/issue/2394#224165)**

Mass-moving tickets not planned for the 1.13 release to 1.14

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2016-02-16 12:12:53](https://pagure.io/SSSD/sssd/issue/2394#224166)**

This would be nice to have, but the changes required for the failover and responder refactoring are not scoped for the 1.14 release.

milestone: SSSD 1.14 beta => SSSD 1.15 beta
sensitive:  => 0

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2017-02-24 14:29:20](https://pagure.io/SSSD/sssd/issue/2394#224167)**

**Metadata Update from @pbrezina**:
- Issue marked as depending on: #1083
- Issue marked as depending on: #2393
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2020-03-13 12:26:48](https://pagure.io/SSSD/sssd/issue/2394#633124)**

**Metadata Update from @thalman**:
- Custom field blocking adjusted to 1083,2393
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue tagged with: Canditate to close

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2020-03-24 14:10:30](https://pagure.io/SSSD/sssd/issue/2394#636586)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [pbrezina](https://pagure.io/user/pbrezina) at [2020-03-24 14:10:31](https://pagure.io/SSSD/sssd/issue/2394#636587)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
